### DISCUSSION TITLE
Async queue in practice

### DISCUSSION THREAD

**Discussion by georgolden** [HowProgrammingWorks/LiveQA#8] [Q&A] [answered]
Практическое применение асинхронной очереди для управления конкуренцией. Когда и для чего ее лучше применять?

**Comment by tshemsedinov** [comment, marked as answer]
Для ограничения пропускной способности есть взаимозаменяемые варианты
- Асинхронная очередь
- Семафор со счетчиком
- Потоки

Выбирать их можно в зависимости от специфики вашего кода и задачи, если нужно ограничить внутри функции кол-во одновременно зашедших в поток исполнения запросов, то это семафор с enter/leave, если на событиях, то или очередь или потои.

**Reply by tshemsedinov** [reply]
Ответ был в записи https://youtu.be/DFBm4cNYaAY

**Reply by georgolden** [reply]
Да, я видел еще тогда

**Reply by tshemsedinov** [reply]
Это я собираю QA для студентов
